vlog4j-core/src/test/java/org/semanticweb/vlog4j/core/reasoner/implementation/LoadDataFromRdfFileTest.java

```diff
@@ -0,0 +1,116 @@
+import org.semanticweb.vlog4j.core.reasoner.FileDataSourceUtils;
+import org.semanticweb.vlog4j.core.reasoner.Reasoner;
+import org.semanticweb.vlog4j.core.reasoner.exceptions.EdbIdbSeparationException;
+import org.semanticweb.vlog4j.core.reasoner.exceptions.IncompatiblePredicateArityException;
+import org.semanticweb.vlog4j.core.reasoner.exceptions.ReasonerStateException;
+
+import karmaresearch.vlog.EDBConfigurationException;
+
+public class LoadDataFromRdfFileTest {
+
+	private static final Predicate ternaryPredicate = Expressions.makePredicate("triple", 3);
+	private static final Atom queryAtom = Expressions.makeAtom(ternaryPredicate, makeVariable("s"), makeVariable("p"),
+			makeVariable("o"));
+
+	@SuppressWarnings("unchecked")
+	private static final Set<List<Term>> expectedTernaryQueryResult = Sets.newSet(
+			Arrays.asList(makeConstant("<c1>"), makeConstant("<q>"), makeConstant("<c2>")),
+			Arrays.asList(makeConstant("<c1>"), makeConstant("<p>"), makeConstant("<c2>")));
+
+	@Test
+	public void testLoadEmptyRdfFile()
+			throws IOException, ReasonerStateException, EdbIdbSeparationException, IncompatiblePredicateArityException {
+		for (final String rdfFileName : Arrays.asList("empty.nt", "empty.nt.gz")) {
+			final FileDataSource emptyFileDataSource = new RdfFileDataSource(
+					new File(FileDataSourceUtils.INPUT_FOLDER + rdfFileName));
+			FileDataSourceUtils.testLoadEmptyFile(ternaryPredicate, queryAtom, emptyFileDataSource);
+		}
+	}
+
+	@Test
+	public void testLoadTernaryFactsFromRdfFile() throws ReasonerStateException, EdbIdbSeparationException,
+	EDBConfigurationException, IOException, IncompatiblePredicateArityException {
+		for (final String rdfFileName : Arrays.asList(FileDataSourceUtils.unzippedNtFileRoot + ".nt",
+				FileDataSourceUtils.zippedNtFileRoot + ".nt.gz")) {
+			final FileDataSource fileDataSource = new RdfFileDataSource(
+					new File(FileDataSourceUtils.INPUT_FOLDER + rdfFileName));
+
+			try (final Reasoner reasoner = Reasoner.getInstance()) {
+				reasoner.addFactsFromDataSource(ternaryPredicate, fileDataSource);
+				reasoner.load();
+
+				final QueryResultIterator queryResultIterator = reasoner.answerQuery(queryAtom, true);
+				final Set<List<Term>> queryResult = QueryResultsUtils.collectQueryResults(queryResultIterator);
+
+				assertEquals(expectedTernaryQueryResult, queryResult);
+			}
+		}
+	}
+
+	@Test(expected = IOException.class)
+	public void testLoadNonexistingRdfFile()
+			throws IOException, ReasonerStateException, EdbIdbSeparationException, IncompatiblePredicateArityException {
+		final File nonexistingFile = new File("nonexistingFile.nt");
+		assertFalse(nonexistingFile.exists());
+		final FileDataSource fileDataSource = new RdfFileDataSource(nonexistingFile);
+
+		try (final Reasoner reasoner = Reasoner.getInstance()) {
+			reasoner.addFactsFromDataSource(ternaryPredicate, fileDataSource);
+			reasoner.load();
+		}
+	}
+
+	@Ignore
```

Comment:
I see, we need to discuss with the VLog people on how to handle this problem.

Reply:
Please add a javadoc to explain why this test is ignored, and a TODO (for example, decide on the expected behavior of VLog)

Reply:
I have discussed with the developers of VLog. The parser ignores lines of an .nt file which do not have a valid N-Triples format. Therefore, the given test file seems to contain no facts over a given ternary predicate (both lines are not N-Triple). This is the expected behavior. I guess this unit test can be modified to test this, instead?